### ISSUE TITLE
Extension doesn't support parent/child configuration

### ISSUE THREAD

**Issue by mgrebenets** [vknabel/vscode-swiftlint#30]
Apparently nested configurations was such a mess in SwiftLint that they completely broke the way it worked before and introduced the new parent/child configuration support.
https://github.com/realm/SwiftLint#child--parent-configs-locally

The extension doesn't support it though.

In my example I had the parent config sitting in the root of my project and a child config sitting few levels down referring back to child config.
For example

```
root
├── .swiftlint.yml
└── dir
    └── dir
        └── .swiftlint.yml
```

The parent config doesn't disable `force_try` which is on by default.
The `dir/dir/.swiflint.yml` child config referes to parent config and disables `force_try` rule for all files sitting in `dir/dir`.

```
parent_config: ../../.swiftlint.yml

disabled_rules:
  - force_try
```

This works good with command line invocations, e.g.

```
cd root
swiftlint .
```

It will respect child configs.

The extension doesn't though.
Initially when the project is loaded it seems to report `force_try` for all files including those in `dir/dir` which should not be reported.
But then, when I actually open the file from `dir/dir` in the editor, it doesn't just report `force_try` but also seems to completely disregard the parent `.swiftlint.yml` from the root and use default SwiftLint ruleset like 120 char line limit and so on.

I tried adding all the `.swiftlint.yml` files to Config Search Paths but it didn't make any difference and I wouldn't really want to add every single `.swiftlint.yml` in the project to the settings, it's difficult to maintain.

**Comment by vknabel** [comment]
Hey @mgrebenets!

Thanks for the great issue description! 

In theory 1.4.0 and #23 should've added nested config support if `swiftlint.configSearchPaths` is empty. But I still found the outdated default value `[".swiftlint.yml"]` within the code base.

Does [1.4.3](https://github.com/vknabel/vscode-swiftlint/releases/tag/1.4.3) fix your issue?

**Comment by mgrebenets** [comment]
As far as I could track it down when listing the entire workspace the extension passes the project root path to SwiftLint, which allows SwiftLint to find parent/child config files.

When going into an editor window it ends up using `--use-stdin` flag with current working directory set to the parent directory of the currently opened file. It also passes the whole file via stdin.

E.g. something like this:

```
cd sub/dir
cat file.swift | swiftlint --use-stdin
```
I suspect it's intended behavior of SwiftLint.
When we pass it Swift code via stdin it shouldn't really care what's the current working directory. We don't pass it any actual path to figure out where to look for config file either.

I understand that extension wants to pass file content rather than the file to have an option to lint currently selected text vs the whole file.

Maybe there's some magic the extension could add?
E.g. look up for for first `.swiftlint.yml` in parent directories within the workspace.
Or maybe always run `swiftlint` command passing the file path.

**Comment by vknabel** [comment]
I can't thank you enough! I actually wrote multiple comments just to delete everything before sending. I always found a new edge case.

When running `swiftlint --use-stdin` we'd need to get into the current file's parent directory which contains the most specific `.swiftlint.yml`-file and then lint. This would lead to mostly correct results.

**But this would break the behaviour of `excluded`**! And even explicitly passing `--config` stops nested configurations from working correctly (and I try to avoid parsing all configs).

Instead I decided to **only lint files after being saved**. Currently it is only possible to view the actually executed commands by cloning this repo and starting the debugger. In the case of this issue the executed command would look something like:

```bash
cd /Users/vknabel/Desktop/Swiftlint &&  SCRIPT_INPUT_FILE_0='dir/dir/Example.swift' SCRIPT_INPUT_FILE_COUNT='1' '/usr/local/bin/swiftlint'  '--use-script-input-files' '--quiet' '--reporter' 'json'
```

The new behaviour is available as [1.4.4](https://github.com/vknabel/vscode-swiftlint/releases/tag/1.4.4). Can you test the new release?

---

In the mid to long term, I try to re-add the lint-while-typing-feature. But I guess this takes a little bit more work.
